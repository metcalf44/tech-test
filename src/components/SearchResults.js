
import React from 'react';
import "../styles/SearchResults.css";
import PropTypes from "prop-types";


const SearchResults = ({ results, searched }) => {

    if (results.length === 0 && searched === false) {
        return null;
    }
    if (results.length === 0 && searched === true) {
        return (
            <p className="search-results-error">No results</p>
        )
    } else {
        return (
        <div className="search-results">
            {results.map((image) => (
                    <img className="card-image"
                    key={image} 
                    src={image} 
                    alt="spaceImage" 
                    data-testid="image"
                    />
            ))}   
        </div>
        )
    }
}
        

SearchResults.propTypes = {
    results: PropTypes.array.isRequired,
};

export default SearchResults