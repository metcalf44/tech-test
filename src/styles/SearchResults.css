
.search-results {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-evenly;
    margin-top: 2.5em;
}

.card-image {
    width: 30%;
    height: 20vh;
    border-radius: 5%;
    margin-bottom: 1em;
}

.search-results-error {
    text-align: center;
    font-weight: bolder;
    font-size: 2.5rem;
    margin-bottom: 3em;
}

@media only screen and (min-width: 760px) and (max-width:1200px) {
    .card-image {
        width: 23%;
        height: 12vh;
    }
}

@media only screen and (min-width: 1201px) {
    .card-image {
        width: 19%;
        height: 10vh;
    }
}
