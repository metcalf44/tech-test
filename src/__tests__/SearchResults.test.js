
import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {

    const images = [
        "https://images-assets.nasa.gov/image/PIA12229/PIA12229~thumb.jpg",
        "https://images-assets.nasa.gov/image/PIA12233/PIA12233~thumb.jpg",
        "https://images-assets.nasa.gov/image/PIA14208/PIA14208~thumb.jpg",
    ];

    const { asFragment } = render(<SearchResults results={images} />);

    it("renders correctly", () => {
        expect(asFragment()).toMatchSnapshot();
    });

    it('displays the number of search results', () => {
        const { getAllByTestId } = render(<SearchResults results={images} />);
        expect(getAllByTestId('image')).toHaveLength(3);
    });
});