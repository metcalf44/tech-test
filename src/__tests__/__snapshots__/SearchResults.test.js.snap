// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <img
      alt="spaceImage"
      class="card-image"
      data-testid="image"
      src="https://images-assets.nasa.gov/image/PIA12229/PIA12229~thumb.jpg"
    />
    <img
      alt="spaceImage"
      class="card-image"
      data-testid="image"
      src="https://images-assets.nasa.gov/image/PIA12233/PIA12233~thumb.jpg"
    />
    <img
      alt="spaceImage"
      class="card-image"
      data-testid="image"
      src="https://images-assets.nasa.gov/image/PIA14208/PIA14208~thumb.jpg"
    />
  </div>
</DocumentFragment>
`;
